server/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java

```diff
@@ -481,7 +510,8 @@ public synchronized void save(Path configDir, char[] password) throws Exception
 
         Directory directory = new NIOFSDirectory(configDir);
         // write to tmp file first, then overwrite
-        String tmpFile = KEYSTORE_FILENAME + ".tmp";
```

Comment:
Be extra careful not to leave temp files around now that `KeystoreWrapper#save` can be invoked from different places, and if some are left around, ensure it doesn't prevent future "save" operations.